### DISCUSSION TITLE
Changing CSS root: variables don't works like expected

### DISCUSSION THREAD

**Discussion by jcenlo** [twbs/bootstrap#37268] [Q&A] [answered]
Hi i am working with bootstrap@5.2.1 i am trying to create  a theme editor without webpack or something similar

my strategy is get the CDN in the document and insert the new variables in a new style tag like this

<img width="1170" alt="Screenshot 2022-10-06 at 16 55 36" src="https://user-images.githubusercontent.com/1485745/194346583-1ac5595e-4969-4c8c-bb7f-53730bb075e4.png">

my intention is rewrite the variables to create a light customisable theme

When i try  change the variables, in some rules works, but not in all
<img width="562" alt="Screenshot 2022-10-06 at 16 58 42" src="https://user-images.githubusercontent.com/1485745/194347368-2f2eea3c-45a0-4730-ad58-4ffdde7bb0cd.png">

For example when i try to change the buttons colors, the rules not contain css var(--var)
<img width="556" alt="Screenshot 2022-10-06 at 16 59 48" src="https://user-images.githubusercontent.com/1485745/194347644-8b7a904e-2a71-4040-9f28-03e8b87ba1c8.png">

for example i expected something like that
<img width="558" alt="Screenshot 2022-10-06 at 17 05 49" src="https://user-images.githubusercontent.com/1485745/194349010-c6d3e4e4-8683-4d54-a734-99a9aa2cfbac.png">

i know this can generate problems with the hover colors, focus, etc... because of the variations of the color

Any way , anyone can help me or is something bootstrap need to review?

**Comment by mdo** [comment, marked as answer]
This is something we can't do yet without breaking a lot of other things. Hopefully that will happen in v6.
